Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,828 @@
+    }
+}
+
+- (BOOL) isProcessedDocument: (NSXMLDocument *)xmlIn
+{
+  NSArray *docArray = [xmlIn nodesForXPath: @"document" error: NULL];
+  if ([docArray count] > 0)
+    {
+      NSXMLElement *docElem = (NSXMLElement *)[docArray objectAtIndex: 0];
+      NSXMLNode *a = [docElem attributeForName: @"processed"];
+      NSString *value = [a stringValue];
+      if (value != nil)
+        {
+          return YES;
+        }
+      else
+        {
+          NSXMLNode *new_attr = [NSXMLNode attributeWithName: @"processed"
+                                                      stringValue: @"true"];
+          [docElem addAttribute: new_attr];
+        }
+    }
+  return NO;
+}
+
+- (NSXMLElement *) createStoryboardProxyElementWithSelector: (NSString *)selector
+                                                     target: (NSString *)dst
+                                            segueIdentifier: (NSString *)ident
+                                                     sender: (NSString *)src
+                                                       kind: (NSString *)kind
+{
+  NSXMLElement *sbproxy = [NSXMLElement elementWithName: @"storyboardSeguePerformAction"];
+
+  NSXMLNode *pselector
+    = [NSXMLNode attributeWithName: @"selector"
+                       stringValue: selector];
+  NSXMLNode *ptarget
+    = [NSXMLNode attributeWithName: @"target"
+                       stringValue: dst];
+  NSString *pident_value = [[NSUUID UUID] UUIDString];
+  NSXMLNode *pident
+    = [NSXMLNode attributeWithName: @"id"
+                       stringValue: pident_value];
+  NSXMLNode *psegueIdent
+    = [NSXMLNode attributeWithName: @"identifier"
+                       stringValue: ident];
+  NSXMLNode *psender
+    = [NSXMLNode attributeWithName: @"sender"
+                       stringValue: src];
+  NSXMLNode *pkind
+    = [NSXMLNode attributeWithName: @"kind"
+                       stringValue: kind];
+  
+  [sbproxy addAttribute: pselector];
+  [sbproxy addAttribute: ptarget];
+  [sbproxy addAttribute: pident];
+  [sbproxy addAttribute: psegueIdent];
+  [sbproxy addAttribute: psender];
+  [sbproxy addAttribute: pkind];
+
+  return sbproxy;
+}
+
+- (NSMapTable *) processConnections: (NSArray *)connectionsArray
+                        withObjects: (NSXMLElement *)objects
+                       controllerId: (NSString *)src
+{
+  NSMapTable *mapTable = [NSMapTable strongToWeakObjectsMapTable];
+  
+  FOR_IN (NSXMLElement*, connObj, connectionsArray)
+    {
+      NSXMLElement *connections = (NSXMLElement *)connObj;
+      NSArray *children = [connections children]; // there should be only one per set.
+      
+      FOR_IN (id, obj, children)
+        {
+          if ([[obj name] isEqualToString: @"segue"])
+            {
+              // get the information from the segue.
+              id segue_parent_parent = [[obj parent] parent];
+              id segue_parent = [obj parent];
```

Comment:
The variable "segue_parent" is always the same as "connections" and "segue_parent_parent" is also the same during one execution of this method.

Reply:
segue_parent_parent should be the object which the segue it belongs to (aside from the connections).  Usually a controller.  This is why there is a check below.

Reply:
I set segue_parent to connections.  I also renamed the "segue_parent_parent" to connections_parent.   The issue here is that it could be "application"... if it is then we want to take certain actions.  I believe this is resolved.